### PR TITLE
negar/account_opening

### PR DESCRIPTION
**Pull request by negarn**
use promise and new form validation

### CODE REVIEW

File: src/javascript/binary/websocket_pages/user/new_account/virtual_acc_opening.js [hunk 1]
```diff
@@ -1,92 +1,115 @@
...
+                true);
+        }
+
+        switch (error.code) {
+            case 'InvalidToken': {
+                const message = 'Your token has expired. Please click <a class="pjaxload" href="[_1]">here</a> to restart the verification process.';
+                return showFormError(message, '');
+            }
+            case 'duplicate email': {
+                const message = 'Your provided email address is already in use by another Login ID. According to our terms and conditions, you may only register once through our site. If you have forgotten the password of your existing account, please <a href="[_1]">try our password recovery tool</a> or contact customer service.';
```
Comment (kellybinary) [inline src/javascript/binary/websocket_pages/user/new_account/virtual_acc_opening.js]: @negarn just a suggestion: 'The email address provided is already in use. If you forgot your password, please try our <a href="[_1]">password recovery tool</a> or contact our <a href="">customer service</a>.'

File: src/javascript/binary/websocket_pages/user/new_account/virtual_acc_opening.js [hunk 2]
```diff
@@ -1,92 +1,115 @@
...
+            return Client.process_new_account(
+                new_account.email,
+                new_account.client_id,
+                new_account.oauth_token,
+                true);
+        }
+
+        switch (error.code) {
+            case 'InvalidToken': {
+                const message = 'Your token has expired. Please click <a class="pjaxload" href="[_1]">here</a> to restart the verification process.';
```
Comment (kellybinary) [inline src/javascript/binary/websocket_pages/user/new_account/virtual_acc_opening.js]: @negarn do we need to use `localize('...')`?

File: src/javascript/binary/websocket_pages/user/new_account/virtual_acc_opening.js (hunk 1 above)
Comment (negarn) [inline src/javascript/binary/websocket_pages/user/new_account/virtual_acc_opening.js]: @ashkanx what do you think? I like it.

File: src/javascript/binary/websocket_pages/user/new_account/virtual_acc_opening.js (hunk 2 above)
Comment (negarn) [inline src/javascript/binary/websocket_pages/user/new_account/virtual_acc_opening.js]: @kellybinary localize is handled inside the `showFormError` function.

**Comment by kellybinary** [comment]
these changes looks good! 👍 👍

File: src/javascript/binary/websocket_pages/user/new_account/virtual_acc_opening.js (hunk 1 above)
Comment (ashkanx) [inline src/javascript/binary/websocket_pages/user/new_account/virtual_acc_opening.js]: Agree.

**Pull request merged**